### PR TITLE
chore(lib): update to JSII 5

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
The snapshot format changed, so there was a lot of partly manual changes. But CI should tell us so please only review the [first commit](https://github.com/hashicorp/terraform-cdk/pull/2776/commits/315d69ca1d039710d2025787d6535dbef126d6fd) :)

### CODE REVIEW

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

File: packages/@cdktf/hcl2cdk/lib/iteration.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function forEachProvider<T, R>(
   return Object.entries(record || {}).reduce((carry, [key, items]) => {
     return {
       ...carry,
-      ...items.reduce((innerCarry, item: T & { alias?: string }) => {
+      ...items.reduce((innerCarry, item: T) => {
```
Comment (ansgarm) [inline packages/@cdktf/hcl2cdk/lib/iteration.ts:44]: ah, nice 👏

File: packages/@cdktf/hcl2json/test/hcl2json.test.ts [hunk 2]
```diff
@@ -93,19 +93,15 @@ describe("convertFiles", () => {
...
   });
 
-  test("invalid files", async () => {
-    try {
-      await convertFiles(path.join(__dirname, "fixtures", "invalid-files"));
-    } catch (e) {
-      expect(e.message).toMatch(/Invalid multi-line string/);
-    }
+  test("invalid files", () => {
+    expect(
```
Comment (ansgarm) [inline packages/@cdktf/hcl2json/test/hcl2json.test.ts:97]: (Blocker)
Mutahhir just recently fixed some tests by returning the result from `expect(...).rejects|resolves` – I think we need to do this here, too.

Related docs: https://jestjs.io/docs/asynchronous#resolves--rejects

File: packages/@cdktf/hcl2json/test/hcl2json.test.ts [hunk 3]
```diff
@@ -93,19 +93,15 @@ describe("convertFiles", () => {
...
   });
 
-  test("invalid path", async () => {
-    try {
-      await convertFiles(path.join("/some/not/existing/path"));
-    } catch (e) {
-      expect(e.message).toMatch(/no such file or directory/);
-    }
+  test("invalid path", () => {
+    expect(convertFiles(path.join("/some/not/existing/path"))).rejects.toMatch(
```
Comment (ansgarm) [inline packages/@cdktf/hcl2json/test/hcl2json.test.ts:103]: (Blocker)
[same here](https://jestjs.io/docs/asynchronous#resolves--rejects)

**Review by ansgarm — CHANGES_REQUESTED** [review]
Nice work! Only two simple blockers (two rewritten tests have a tiny oversight that makes them more flaky) – the only reason why I didn't approve 😄

File: packages/@cdktf/cli-core/package.json [hunk 4]
```diff
@@ -38,13 +38,13 @@
...
-    "codemaker": "^1.76.0",
+    "codemaker": "^1.79.0",
     "constructs": "^10.0.25",
     "cross-spawn": "^7.0.3",
     "https-proxy-agent": "^5.0.1",
     "ink-select-input": "^4.2.1",
-    "jsii": "^1.77.0",
-    "jsii-pacmak": "^1.77.0",
+    "jsii": "^5.0.1",
+    "jsii-pacmak": "^1.79.0",
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply packages/@cdktf/cli-core/package.json:47]: I think whatever is available, afaik. there is no newer version released

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

**Review by ansgarm — APPROVED** [review]

**Pull request merged**